src/components/calendar/calendar.jsx

```diff
@@ -42,32 +43,146 @@ function getCellClassName(iterator, month, highlightedDate) {
+    const ref = React.createRef();
+
+    const cellRef = {};
+    cellRef[iterator.toDateString()] = ref;
+    refs = { ...cellRef, ...refs };
+
+    const sameDate = iterator.getDate() === focusedDate.getDate() && iterator.getMonth() === focusedDate.getMonth();
+    const tabIndex = sameDate ? 0 : null;
+
+    // This needs to occur at the end of this function
+    iterator.setDate(date + 1); return (
+      <td tabIndex={tabIndex} className={className} ref={ref} >
+        {date}
+      </td>
+    );
+  };
+
+  useEffect(() => {
+    const ref = refs[focusedDate.toDateString()];
+    if (ref) {
+      ref.current.focus();
+    }
+  }, [focusedDate]);
+
+  function handleKeyboardYearChange(direction) {
+    changeMonth(12 * direction);
+    const newDate = new Date(focusedDate.getFullYear() + direction, focusedDate.getMonth(), focusedDate.getDate());
+    setFocusedDate(newDate);
+  }
+
+  function handleKeyboardDateChange(dateAmount) {
+    const newDate = new Date(focusedDate.setDate(focusedDate.getDate() + dateAmount));
+    setFocusedDate(newDate);
+    const ref = refs[newDate.toDateString()];
+    if (!ref) changeMonth(Math.abs(dateAmount) / dateAmount);
+  }
+
+  function onKeyDown(event) {
+    switch (event.key) {
+      case 'ArrowLeft':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(-1);
+        }
+        break;
+      case 'ArrowUp':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(-7);
+        }
+        break;
+      case 'ArrowRight':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(1);
+        }
+        break;
+      case 'ArrowDown':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(7);
+        }
+        break;
+      case 'End':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(6 - focusedDate.getDay());
+        }
+        break;
+      case 'Home':
+        if (active) {
+          event.preventDefault();
+          handleKeyboardDateChange(0 - focusedDate.getDay());
+        }
+        break;
+      case 'PageUp':
+        if (active) {
+          event.preventDefault();
+          if (event.shiftKey) {
+            handleKeyboardYearChange(-1);
+          } else {
+            changeMonth(-1);
+            handleKeyboardDateChange(-getDaysInMonth(focusedDate, 0));
+          }
+        }
+        break;
+      case 'PageDown':
+        if (active) {
+          event.preventDefault();
+          if (event.shiftKey) {
+            handleKeyboardYearChange(1);
+          } else {
+            changeMonth(1);
+            handleKeyboardDateChange(getDaysInMonth(focusedDate, 1));
+          }
+        }
+        break;
+      default:
+    }
+  }
+
+  function onFocus(event) {
+    const activeRef = Object.values(refs).find(ref => ref.current.contains(event.target));
```

Comment:
If we're concerned about body cells receiving focus, we should just set the focus handler on the `<tbody>` instead of the table.

Reply:
This is just making sure we arn't focusing anything completely outside the component. So I think we would need this regardless of whether the focus was on `tbody` or `table`.

Reply:
The focus event won't be handled by this `onFocus` if the focus event originated from outside the component. If that was the case for this code then we can remove this.